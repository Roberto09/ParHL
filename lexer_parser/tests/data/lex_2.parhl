let fibo (n : int) : int{
    if(n = 1 or n = 2){
        return n-1;
    }
    return n + fibo(n-1) + fibo(n-2);
}

fibo(10)
