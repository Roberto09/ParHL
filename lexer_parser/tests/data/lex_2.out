Token(type='LET', value='let', lineno=1, index=0)
Token(type='ID', value='fibo', lineno=1, index=4)
Token(type='L_PAREN', value='(', lineno=1, index=9)
Token(type='ID', value='n', lineno=1, index=10)
Token(type='COLON', value=':', lineno=1, index=12)
Token(type='INT_T', value='int', lineno=1, index=14)
Token(type='R_PAREN', value=')', lineno=1, index=17)
Token(type='COLON', value=':', lineno=1, index=19)
Token(type='INT_T', value='int', lineno=1, index=21)
Token(type='L_BRACE', value='{', lineno=1, index=24)
Token(type='NEWLINE', value='\n', lineno=1, index=25)
Token(type='IF', value='if', lineno=2, index=30)
Token(type='L_PAREN', value='(', lineno=2, index=32)
Token(type='ID', value='n', lineno=2, index=33)
Token(type='EQ', value='=', lineno=2, index=35)
Token(type='INT_V', value=(1, 'INT_V'), lineno=2, index=37)
Token(type='OR', value='or', lineno=2, index=39)
Token(type='ID', value='n', lineno=2, index=42)
Token(type='EQ', value='=', lineno=2, index=44)
Token(type='INT_V', value=(2, 'INT_V'), lineno=2, index=46)
Token(type='R_PAREN', value=')', lineno=2, index=47)
Token(type='L_BRACE', value='{', lineno=2, index=48)
Token(type='NEWLINE', value='\n', lineno=2, index=49)
Token(type='RETURN', value='return', lineno=3, index=58)
Token(type='ID', value='n', lineno=3, index=65)
Token(type='MINUS', value='-', lineno=3, index=66)
Token(type='INT_V', value=(1, 'INT_V'), lineno=3, index=67)
Token(type='SEMICOLON', value=';', lineno=3, index=68)
Token(type='NEWLINE', value='\n', lineno=3, index=69)
Token(type='R_BRACE', value='}', lineno=4, index=74)
Token(type='NEWLINE', value='\n', lineno=4, index=75)
Token(type='RETURN', value='return', lineno=5, index=80)
Token(type='ID', value='n', lineno=5, index=87)
Token(type='PLUS', value='+', lineno=5, index=89)
Token(type='ID', value='fibo', lineno=5, index=91)
Token(type='L_PAREN', value='(', lineno=5, index=95)
Token(type='ID', value='n', lineno=5, index=96)
Token(type='MINUS', value='-', lineno=5, index=97)
Token(type='INT_V', value=(1, 'INT_V'), lineno=5, index=98)
Token(type='R_PAREN', value=')', lineno=5, index=99)
Token(type='PLUS', value='+', lineno=5, index=101)
Token(type='ID', value='fibo', lineno=5, index=103)
Token(type='L_PAREN', value='(', lineno=5, index=107)
Token(type='ID', value='n', lineno=5, index=108)
Token(type='MINUS', value='-', lineno=5, index=109)
Token(type='INT_V', value=(2, 'INT_V'), lineno=5, index=110)
Token(type='R_PAREN', value=')', lineno=5, index=111)
Token(type='SEMICOLON', value=';', lineno=5, index=112)
Token(type='NEWLINE', value='\n', lineno=5, index=113)
Token(type='R_BRACE', value='}', lineno=6, index=114)
Token(type='NEWLINE', value='\n', lineno=6, index=115)
Token(type='NEWLINE', value='\n', lineno=7, index=116)
Token(type='ID', value='fibo', lineno=8, index=117)
Token(type='L_PAREN', value='(', lineno=8, index=121)
Token(type='INT_V', value=(10, 'INT_V'), lineno=8, index=122)
Token(type='R_PAREN', value=')', lineno=8, index=124)
Token(type='NEWLINE', value='\n', lineno=8, index=125)
